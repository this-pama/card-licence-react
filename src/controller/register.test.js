import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import RequestApp from './register';

describe('test request App page', () => {
  it('renders without crashing', () => {
    shallow(<RequestApp />);
  });
});

describe('test request app snapshot', () => {
  it('test snapshot', () => {
    const component = renderer.create(<RequestApp />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.handleEmail;
    tree.props.handleFullName;
    tree.props.handleUsername;
    tree.props.handleDevice;
    tree.props.handleAddress;
    tree.props.handlePhoneNumber;
    tree.props.validateForm;
    tree.props.request;
    tree.props.email;
    tree.props.fullName;
    tree.props.device;
    tree.props.username;
    tree.props.phoneNumber;
    tree.props.address;
    tree.props.spinner;
    tree.props.errMessage;
    tree.props.disable;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('simulate events', () => {
  const wrapper = mount(<RequestApp />);

  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = mount(<RequestApp licence={onButtonClick} />);
    wrapper.find('Button').simulate('click');
    // expect(onButtonClick).toHaveReturnedWith('errMessage', '');
  });
});

describe('Test internal states of methods', () => {
  const component = renderer.create(<RequestApp />);
  const instance = component.getInstance();

  test('it change state', () => {
    expect(instance.state.fullName).toBe('');
    expect(instance.state.device).toBe('');
    expect(instance.state.address).toBe('');
    expect(instance.state.phoneNumber).toBe('');
    expect(instance.state.username).toBe('');
    expect(instance.state.email).toBe('');
    expect(instance.state.errMessage).toBe('');
    expect(instance.state.spin).toBe(false);
    const value = { target: { value: 'foo' } };
    instance.handleEmail(value);
    expect(instance.state.email).toBe('foo');
    instance.handleFullName(value);
    expect(instance.state.fullName).toBe('foo');
    instance.handleDevice(value);
    expect(instance.state.device).toBe('foo');
    instance.handleUsername(value);
    expect(instance.state.username).toBe('foo');
    instance.handleAddress(value);
    expect(instance.state.address).toBe('foo');
    instance.handlePhoneNumber(value);
    expect(instance.state.phoneNumber).toBe('foo');
  });

  test('validate form state', () => {
    instance.validateForm();
    expect(instance.state.disable).toBe(true);
  });

  test('spinner state', () => {
    instance.request();
    expect(instance.state.spin).toBe(true);
  });
});
