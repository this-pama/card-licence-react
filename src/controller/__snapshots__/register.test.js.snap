// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test request app snapshot test snapshot 1`] = `
<div
  style={
    Object {
      "backgroundImage": "logo.svg",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-2"
      />
      <div
        className="col-md-6 col-sm-8"
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "border": 1,
                "color": "white",
              }
            }
          >
            <h3
              style={
                Object {
                  "padding": 15,
                  "textAlign": "center",
                }
              }
            >
              Card Licence Request
            </h3>
          </div>
          <p
            style={
              Object {
                "color": "red",
                "paddingTop": 10,
              }
            }
          >
            
          </p>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Full Name
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Username
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Device
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Device"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Phone Number
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Phone Number"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Email address
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />
            <small
              className="text-muted form-text"
            >
              We'll never share your email with anyone else.
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Address
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Address"
              type="text"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-block"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
             Make Request
          </button>
        </div>
      </div>
      <div
        className="col-md-3 col-sm-2"
      />
    </div>
  </div>
</div>
`;

exports[`test request app snapshot test snapshot 2`] = `
<div
  style={
    Object {
      "backgroundImage": "logo.svg",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3 col-sm-2"
      />
      <div
        className="col-md-6 col-sm-8"
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "border": 1,
                "color": "white",
              }
            }
          >
            <h3
              style={
                Object {
                  "padding": 15,
                  "textAlign": "center",
                }
              }
            >
              Card Licence Request
            </h3>
          </div>
          <p
            style={
              Object {
                "color": "red",
                "paddingTop": 10,
              }
            }
          >
            
          </p>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Full Name
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Username
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Device
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Device"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Phone Number
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Phone Number"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Email address
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />
            <small
              className="text-muted form-text"
            >
              We'll never share your email with anyone else.
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Address
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Address"
              type="text"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-block"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
             Make Request
          </button>
        </div>
      </div>
      <div
        className="col-md-3 col-sm-2"
      />
    </div>
  </div>
</div>
`;
